code/datums/components/team_monitor.dm

```diff
@@ -0,0 +1,535 @@
+		screen = new
+		screen.alpha = 240
+		screen.color = beacon.colour
+		screen.hud = updating.hud_used
+		updating.hud_used.team_finder_arrows += screen
+		tracking[beacon] = screen
+		//Update their hud
+		updating.hud_used.show_hud(updating.hud_used.hud_version, updating)
+	var/matrix/rotationMatrix = matrix()
+	rotationMatrix.Scale(1.5)
+	rotationMatrix.Translate(0, -distance)
+	rotationMatrix.Turn(Get_Angle(target_turf, parent_turf))
+	animate(screen, transform = rotationMatrix, time = 2)
+
+//===========
+// Handles being equipped / dequipped
+//===========
+
+//The parent equipped an item with a team_monitor, check if its in the right slot and apply the hud
+//Also needs to enable other trackers pointers towards us
+/datum/component/team_monitor/proc/parent_equipped(datum/source, mob/equipper, slot)
+	var/obj/item/clothing/item = parent
+	if(!istype(item))
+		return
+	if(item.slot_flags & slotdefine2slotbit(slot)) //Was equipped to a valid slot for this item?
+		show_hud(equipper)
+	else
+		hide_hud(equipper)
+
+//Disable our hud
+//Disable the pointers to us
+/datum/component/team_monitor/proc/parent_dequpped(datum/source, mob/user)
+	hide_hud(user)
+
+//===========
+// Handles hiding / showing the hud when equipped
+//===========
+
+/datum/component/team_monitor/proc/show_hud(mob/target)
+	updating = target
+	//Our hud is disabled
+	if(!hud_visible)
+		return
+	//Start processing to update in weird situations
+	START_PROCESSING(SSprocessing, src)
+	//Register parent signal
+	RegisterSignal(target, COMSIG_MOVABLE_MOVED, .proc/parent_moved)
+	//Mob doesnt have a hud, dont add hud arrows
+	if(!target.hud_used)
+		return
+	for(var/datum/component/tracking_beacon/key in tracking)
+		var/atom/movable/screen/arrow/arrow = new
+		arrow.alpha = 240
+		arrow.color = key.colour
+		arrow.hud = target.hud_used
+		target.hud_used.team_finder_arrows += arrow
+		tracking[key] = arrow
+	//Update their hud
+	target.hud_used.show_hud(target.hud_used.hud_version, target)
+	update_all_directions()
+
+/datum/component/team_monitor/proc/hide_hud(mob/target)
+	updating = null
+	//Stop processing
+	STOP_PROCESSING(SSprocessing, src)
+	//UnRegister parent signal
+	UnregisterSignal(target, COMSIG_MOVABLE_MOVED)
+	//Remove our arrows
+	for(var/key in tracking)
+		var/atom/movable/screen/arrow = tracking[key]
+		if(!arrow)
+			continue
+		if(target.hud_used)
+			target.hud_used.team_finder_arrows -= arrow
+		qdel(arrow)
+		tracking[key] = null
+	//Update their hud
+	if(target.hud_used)
+		target.hud_used.show_hud(target.hud_used.hud_version, target)
+
+//===========
+// Handles user interaction
+// - Disabling hud transmission
+// - Disabling hud view
+// - Changing transmission frequency
+//===========
+
+/datum/component/team_monitor/proc/toggle_hud(new_hud_status, mob/user)
+	hud_visible = new_hud_status
+	if(hud_visible && !updating)
+		show_hud(user)
+	else if(!hud_visible)
+		hide_hud(user)
+
+/datum/component/team_monitor/proc/change_frequency(mob/user)
```

Comment:
This proc is not used. Beacons have an item action to `change_frequency`, but not monitors. You made beacon's `change_frequency` to change attached monitor's frequency with `set_frequency`, so you have two options:

1. Remove this proc. There currently isn't a monitor without attached beacon anyway.
2. Make separate item action to change monitor's frequency, ideally with a more complex input system like enabling the choice to (beacon) GLOB frequency and making frequencies synchronize with each other on demand.

Reply:
This proc exists for items that may be implemented in the future that contain a monitor but not a beacon. Without it, it means that it would have to be recoded.

Reply:
I wasn't 100% content with the current state of that code; it looks as if it was a leftover code. For example, monitor's `change_frequency` doesn't set frequency of `attached_beacon`. Obviously, you decide. If *you* are content with the current state of code, then I'll gloss over this bit. However, I'd like you to add a comment about it if this proc really is intended for beacon-less monitors.

Reply:
I guess checking for an attached_beacon and changing that too would make sense for consistency sake